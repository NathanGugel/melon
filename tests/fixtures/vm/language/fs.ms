
let funcs = [
    "exists",
    "isFile",
    "isDirectory",
    "isReadable",
    "isWritable"
];

func listPerms |path| => {
    for (let f in funcs) {
        io.print(f .. "(" .. path .. ")" .. " = " .. string.toString(fs[f](path)));
    }
}

listPerms("wrong");
listPerms("fixtures/vm/language/fs_test_files");
listPerms("fixtures/vm/language/fs_test_files/a_file.txt");