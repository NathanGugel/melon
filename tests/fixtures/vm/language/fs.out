exists(wrong) = false
isFile(wrong) = false
isDirectory(wrong) = false
isReadable(wrong) = false
isWritable(wrong) = false
exists(fixtures/vm/language/fs_test_files) = true
isFile(fixtures/vm/language/fs_test_files) = false
isDirectory(fixtures/vm/language/fs_test_files) = true
isReadable(fixtures/vm/language/fs_test_files) = true
isWritable(fixtures/vm/language/fs_test_files) = true
exists(fixtures/vm/language/fs_test_files/a_file.txt) = true
isFile(fixtures/vm/language/fs_test_files/a_file.txt) = true
isDirectory(fixtures/vm/language/fs_test_files/a_file.txt) = false
isReadable(fixtures/vm/language/fs_test_files/a_file.txt) = true
isWritable(fixtures/vm/language/fs_test_files/a_file.txt) = true
